addon/core/say-data-factory.ts

```diff
@@ -53,8 +56,13 @@ export class SayDataFactory extends DataFactory {
   contentLiteral(
     languageOrDataType?: string | SayNamedNode<string>,
   ): ContentLiteralTerm {
-    const literal = this.literal('', languageOrDataType);
-    return { ...literal, value: '', termType: 'ContentLiteral' };
+    // TODO: we need to be careful with using spread operators here:
```

Comment:
I wonder if it's better to modify the return to include an `equals` method (ideally on the prototype, so it's as similar to the other Terms as possible). The classes for [Literals](https://github.com/rubensworks/rdf-data-factory.js/blob/master/lib/Literal.ts) and [NamedNodes](https://github.com/rubensworks/rdf-data-factory.js/blob/master/lib/NamedNode.ts) seem almost trivial, so we could implement our own versions which compare the correct `termType`s easily.

Reply:
Yes that could be better indeed. We could define our custom terms as classes (not as interfaces). We could even define the `equals` method as an arrow function, so it is preserved when using the spread operator. If we use classes, we are sure we don't accidentally make spreading 'mistakes'.

We could base our implementation on https://www.npmjs.com/package/rdf-data-factory

Reply:
I created a PR to introduce custom term classes: https://github.com/lblod/ember-rdfa-editor/pull/1171. Not entirely sure if it is really necessary, but it should provide a solution to the use of `equals` methods an object spreading.